addons/maintenance/tests/test_maintenance.py

```diff
@@ -78,3 +78,18 @@ def test_10_equipment_request_category(self):
         # I check that maintenance request is in the "In Progress" stage
         self.assertEqual(maintenance_request_01.stage_id.id, self.ref('maintenance.stage_1'))
 
+    def test_forever_maintenance_repeat_type(self):
+        """
+        Test that a maintenance request with repeat_type = forever will be duplicated when it
+        is moved to a 'done' stage, and the new request will be placed in the first stage.
+        """
+        maintenance_request = self.env['maintenance.request'].create({
+            'name': 'Test forever maintenance',
+            'repeat_type': 'forever',
+            'maintenance_type': 'preventive',
+            'recurring_maintenance': True,
+        })
+        maintenance_request.stage_id = self.env['maintenance.stage'].search([('done', '=', True)], limit=1)
+        self.assertTrue(maintenance_request.stage_id.done)
+        new_maintenance = self.env['maintenance.request'].search([('name', '=', 'Test forever maintenance'), ('stage_id.done', '=', False)])
+        self.assertTrue(new_maintenance)
```

Comment:
Quoting your own description:
> Test [...] the new request will be placed in the first stage.

Would be nice to assert that it is indeed the first one, not just a non `done` stage.  :slightly_smiling_face: